src/convert/mapEvent.js

```diff
@@ -1,29 +1,44 @@
-const Y = require('yjs');
-const { toSlatePath } = require('../utils/convert');
+const Y = require("yjs");
+const { Block } = require("slate");
+const { toSlatePath } = require("../utils/convert");
 
-// type MapAction = { action: 'add' | 'update' | 'delete'; oldValue: any };
-// type SetNodeOperationProperties = Pick<
-//   SetNodeOperation,
-//   'newProperties' | 'properties'
-// >;
+const mapInsertNodeOperations = (event) => {
+  const changes = Array.from(event.changes.keys.entries());
+  const operations = [];
+  changes.forEach(([key]) => {
+    if (key === "document") {
```

Comment:
I'm not fully understanding the expectations around the structure of `event` here -- but it feels like the rest of this code reads as if any change that has a key of document will necessarily an insertion?

Reply:
Yep, there's more metadata indicating that the change is an insertion. So it's possible for Yjs to yield remove (and maybe update?) events. I intentionally didn't cover those cases as it's not being used currently by our implementation – we only modify the top level `Y.Map` on document load, all other changes are applied directly to the nested `Y.Array` document.

But now I think I should be at least throwing an error if we see what I said will never happen, it will help debugging if needs be. Will update the code.